src/core/backbone_router/local.cpp

```diff
@@ -137,9 +137,15 @@ void Local::GetConfig(BackboneRouterConfig &aConfig) const
     aConfig.mMlrTimeout          = mMlrTimeout;
 }
 
-void Local::SetConfig(const BackboneRouterConfig &aConfig)
+otError Local::SetConfig(const BackboneRouterConfig &aConfig)
 {
-    bool update = false;
+    otError error  = OT_ERROR_NONE;
+    bool    update = false;
+
+    // Validate configuration according to Thread 1.2.1 Specification 5.21.3.3:
+    // "The Reregistration Delay in seconds MUST be lower than (0.5 * MLR Timeout). It MUST be at least 1."
+    VerifyOrExit(aConfig.mReregistrationDelay >= 1, error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(aConfig.mReregistrationDelay * 2 < aConfig.mMlrTimeout, error = OT_ERROR_INVALID_ARGS);
```

Comment:
Great catch!

However with more analysis, I found that the original condition is actually correct because `mReregistrationDelay` is of type `uint16_t`. In the expression, `mReregistrationDelay` will be promoted to `uint32_t` so it will never overflow. 

That being said, this expression is a potential bug if one day we changed `mReregistrationDelay` to be of type `uint32_t`. In such case, the suggested conditions are however incorrect. I tend to keep the code unchanged since it's already correct.

Reply:
I found it interesting that it's actually difficult to write such a simple expression correctly. 

I think this expression is correct no matter `mReregistrationDelay` is of type `uint16_t` or `uint32_t`. 
```c++
    VerifyOrExit(static_cast<uint32_t>(aConfig.mReregistrationDelay) <= std::numeric_limits<uint32_t>::max() / 2 &&
                     aConfig.mReregistrationDelay < (aConfig.mMlrTimeout + 1) / 2,
                 error = OT_ERROR_INVALID_ARGS);
```

Thoughts? @jwhui

Reply:
Rather than complicating run-time check, how about we use `static_assert` to ensure the types are appropriate for the given logic?

Reply:
I have added the `static_assert` check.